app/src/org/commcare/CommCareApplication.java

```diff
@@ -240,8 +240,12 @@ public void onCreate() {
         LocalePreferences.saveDeviceLocale(Locale.getDefault());
     }
 
+    public static boolean isRoboUnitTest() {
+        return "robolectric".equals(Build.FINGERPRINT);
+    }
+
     private void initTls12IfNeeded() {
-        if (Build.VERSION.SDK_INT >= 16 && Build.VERSION.SDK_INT < 20) {
+        if (Build.VERSION.SDK_INT >= 16 && Build.VERSION.SDK_INT < 20 && !isRoboUnitTest()) {

```

Comment:
Wondering if we can override this function in `CommCareTestApplication` instead ?

Reply:
Ah, clever idea. I was racing to get this in on the hotfix, but that makes **way** more sense. I'll make this change in the long-term version.